rotkehlchen/db/history_events.py

```diff
@@ -117,20 +117,31 @@ def add_history_event(
         the DB. Can only happen if an event with an unresolved asset is passed.
         """
         identifier = None  # overwritten by first write
-        for idx, (insertquery, _, bindings) in enumerate(event.serialize_for_db()):
-            if idx == 0:
-                write_cursor.execute(f'INSERT OR IGNORE INTO {insertquery}', bindings)
-                if write_cursor.rowcount == 0:
-                    return None  # already exists
-                identifier = write_cursor.lastrowid  # keep identifier to use in next insertions
-            else:
-                write_cursor.execute(f'INSERT OR IGNORE INTO {insertquery}', (identifier, *bindings))  # noqa: E501
+        event_tuple = event.serialize_for_db()
+        try:
+            for idx, (insertquery, _, bindings) in enumerate(event_tuple):
+                if idx == 0:
+                    write_cursor.execute(f'INSERT OR IGNORE INTO {insertquery}', bindings)
+                    if write_cursor.rowcount == 0:
+                        return None  # already exists
+                    identifier = write_cursor.lastrowid  # keep identifier to use in next insertions  # noqa: E501
+                else:
+                    write_cursor.execute(f'INSERT OR IGNORE INTO {insertquery}', (identifier, *bindings))  # noqa: E501
 
-        if mapping_values is not None:
-            write_cursor.executemany(
-                'INSERT OR IGNORE INTO history_events_mappings(parent_identifier, name, value) '
-                'VALUES(?, ?, ?)',
-                [(identifier, k, v) for k, v in mapping_values.items()],
+            if mapping_values is not None:
+                write_cursor.executemany(
+                    'INSERT OR IGNORE INTO history_events_mappings(parent_identifier, name, value) '  # noqa: E501
+                    'VALUES(?, ?, ?)',
+                    [(identifier, k, v) for k, v in mapping_values.items()],
+                )
+        except OverflowError:
+            self.db.msg_aggregator.add_error(
```

Comment:
I don't think this is a good idea.

Because before the error was bubbling up and at the destruction of the write context also reverts the DB changes. The way you do this won't revert them. But just log an error.